matrix_sdk_base/src/models/room.rs

```diff
@@ -617,14 +625,33 @@ impl Room {
     /// Returns true if `MessageQueue` was added to.
     #[cfg(feature = "messages")]
     #[cfg_attr(docsrs, doc(cfg(feature = "messages")))]
-    pub fn handle_message(&mut self, event: &MessageEvent) -> bool {
+    pub fn handle_message(&mut self, event: &AnyMessageEventStub) -> bool {
         self.messages.push(event.clone())
     }
 
+    /// Handle a room.redaction event and update the `MessageQueue` if necessary.
+    ///
+    /// Returns true if `MessageQueue` was updated.
+    #[cfg(feature = "messages")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "messages")))]
+    pub fn handle_redaction(&mut self, event: &RedactionEventStub) -> bool {
+        if let Some(msg) = self
+            .messages
+            .iter_mut()
+            .find(|msg| &event.redacts == msg.event_id())
+        {
+            // TODO make msg an enum or use AnyMessageEventStub enum to represent
+            *msg = MessageWrapper(AnyMessageEventStub::RoomRedaction(event.clone()));
```

Comment:
This replaces the existing event with a redaction event, no? This isn't quite right the existing event should be turned into a redacted event and the redaction event should be pushed to the message queue either unconditionally or perhaps not at all since nobody will want to display redaction events.

Or to put it differently, the existing event will be something like a `m.room.message` usually and you're replacing it with a `m.room.redaction`, the event type shouldn't change, only the content of the `m.room.message` event should be gone.

Reply:
I still have to deal with this issue.

Reply:
I followed the spec as much as I could https://matrix.org/docs/spec/client_server/r0.6.1#redactions